### PR TITLE
Implement TableScanNode::addDetails

### PR DESCRIPTION
**Pull request by majetideepak**
Add ConnectorTableHandle::toString and use it in the implementation of TableScanNode::addDetails

The TableScan node can now display the table name and the pushed down filters.

Example for TPC-H q6 query:

```
 -> TableScan[Name: lineitem, Filters: [(l_quantity, DoubleRange: (-inf, 24.000000) no nulls), (l_discount, DoubleRange: [0.050000, 0.070000] no nulls), (l_shipdate, BytesRange: [1994-01-01, 1994-12-31] no nulls), ]]
```

### CODE REVIEW

**Comment by majetideepak** [comment]
This will need a follow-up change on the presto-cpp side. I will add a PR there when this is approved.

**Comment by majetideepak** [comment]
The Presto protocol does provide these values. So should be straightforward to implement a fully qualified name.
```
struct HiveTableHandle : public ConnectorTableHandle {
  String schemaName = {};
  String tableName = {};
  std::shared_ptr<List<List<String>>> analyzePartitionValues = {};

  HiveTableHandle() noexcept;
};
```

File: velox/connectors/Connector.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ class ColumnHandle {
 class ConnectorTableHandle {
  public:
   virtual ~ConnectorTableHandle() = default;
+  virtual const std::string& name() const = 0;
```
Comment (mbasmanova) [inline velox/connectors/Connector.h:63]: Should we add a more generic toString() method instead?

**Review by mbasmanova — COMMENTED** [review]
It is nice to have a bit more details for the TableScan node.

File: velox/connectors/Connector.h (hunk 1 above)
Comment (majetideepak) [reply velox/connectors/Connector.h:63]: good idea!

File: velox/connectors/hive/HiveConnector.cpp [hunk 2]
```diff
@@ -40,6 +40,25 @@ static const char* kPath = "$path";
 static const char* kBucket = "$bucket";
 } // namespace
 
+std::string HiveTableHandle::toString() const {
+  std::stringstream out;
+  out << "Name: " << name_;
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveConnector.cpp:45]: nit: "Table" might be more descriptive

File: velox/connectors/hive/HiveConnector.cpp [hunk 3]
```diff
@@ -40,6 +40,25 @@ static const char* kPath = "$path";
...
+  if (!subfieldFilters_.empty()) {
+    out << ", Filters: [";
+    bool notFirstFilter = false;
+    for (auto& pair : subfieldFilters_) {
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveConnector.cpp:49]: nit: const auto&

File: velox/connectors/hive/HiveConnector.h [hunk 4]
```diff
@@ -63,10 +63,12 @@ using SubfieldFilters =
 class HiveTableHandle : public ConnectorTableHandle {
  public:
   HiveTableHandle(
+      const std::string& name,
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveConnector.h:66]: nit: perhaps, tableName for clarity

File: velox/connectors/hive/HiveConnector.h [hunk 5]
```diff
@@ -82,7 +84,10 @@ class HiveTableHandle : public ConnectorTableHandle {
     return remainingFilter_;
   }
 
+  std::string toString() const;
+
  private:
+  const std::string name_;
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveConnector.h:90]: nit: tableName_ for clarity

File: velox/exec/tests/PlanNodeToStringTest.cpp [hunk 6]
```diff
@@ -408,3 +414,29 @@ TEST_F(PlanNodeToStringTest, exchange) {
...
   ASSERT_EQ("-> Exchange[]\n", plan->toString(true, false));
 }
+
+TEST_F(PlanNodeToStringTest, tableScan) {
+  const auto shipDate = "shipdate";
+  const auto lineitem = "lineitem";
+  RowTypePtr rowType{
+      ROW({"discount", "tax", "shipdate"}, {DOUBLE(), DOUBLE(), VARCHAR()})};
+  ColumnHandleMap assignments = {
+      {"discount", HiveConnectorTestBase::regularColumn("discount", DOUBLE())},
```
Comment (mbasmanova) [inline velox/exec/tests/PlanNodeToStringTest.cpp:424]: nit: HiveConnectorTestBase::allRegularColumns(rowType)

File: velox/exec/tests/PlanNodeToStringTest.cpp [hunk 7]
```diff
@@ -408,3 +414,29 @@ TEST_F(PlanNodeToStringTest, exchange) {
...
+      {"tax", HiveConnectorTestBase::regularColumn("tax", DOUBLE())},
+      {"shipdate",
+       HiveConnectorTestBase::regularColumn("shipdate", VARCHAR())}};
+  auto filters = SubfieldFiltersBuilder()
+                     .add(shipDate, between("1994-01-01", "1994-12-31"))
+                     .add("l_discount", betweenDouble(0.05, 0.07))
+                     .add("l_quantity", lessThanDouble(24.0))
+                     .build();
+  auto tableHandle = std::make_shared<connector::hive::HiveTableHandle>(
```
Comment (mbasmanova) [inline velox/exec/tests/PlanNodeToStringTest.cpp:433]: nit: HiveConnectorTestBase::makeTableHandle

File: velox/exec/tests/utils/TpchQueryBuilder.cpp [hunk 8]
```diff
@@ -91,10 +91,11 @@ using ColumnHandleMap =
     std::unordered_map<std::string, std::shared_ptr<connector::ColumnHandle>>;
 
 std::shared_ptr<connector::hive::HiveTableHandle> makeTableHandle(
+    const std::string& name,
```
Comment (mbasmanova) [inline velox/exec/tests/utils/TpchQueryBuilder.cpp:94]: tableName?

**Review by mbasmanova — APPROVED** [review]
@majetideepak This is nice. Having both table name and a list of pushed down filters printed out is useful. I only have a few nits.

**Comment by majetideepak** [comment]
@mbasmanova thank you for your feedback. I added those changes. I will add the corresponding changes for the presto_cpp side in a bit.

**Comment by mbasmanova** [comment]
@majetideepak Deepak, linux-parquet-s3-build job failed. Would you take a look?

**Comment by majetideepak** [comment]
@mbasmanova Here is the presto_cpp side PR to match this change. 
https://github.com/facebookexternal/presto_cpp/pull/677

File: velox/exec/tests/PlanNodeToStringTest.cpp [hunk 9]
```diff
@@ -14,12 +14,18 @@
...
 #include "velox/exec/tests/utils/PlanBuilder.h"
 #include "velox/functions/prestosql/registration/RegistrationFunctions.h"
 #include "velox/parse/TypeResolver.h"
+#include "velox/type/tests/FilterBuilder.h"
+#include "velox/type/tests/SubfieldFiltersBuilder.h"
 #include "velox/vector/tests/VectorTestBase.h"
 
 using namespace facebook::velox;
+using namespace facebook::velox::common::test;
+using namespace facebook::velox::exec::test;
```
Comment (mbasmanova) [inline velox/exec/tests/PlanNodeToStringTest.cpp:27]: Got an error:

There are multiple using directives that potentially have name conflicts or are redundant.
This using test directive has potential name conflicts with the using test directive at velox/exec/tests/PlanNodeToStringTest.cpp:26:42.
Redundant using declarations may be a sign of code that should be checked for errors.

File: velox/connectors/hive/HiveConnector.h [hunk 10]
```diff
@@ -82,7 +84,10 @@ class HiveTableHandle : public ConnectorTableHandle {
     return remainingFilter_;
   }
 
+  std::string toString() const;
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveConnector.h:87]: Got an error:

Overrides should be marked explicitly.
toString overrides methods from ConnectorTableHandle but is not annotated with override.

**Comment by majetideepak** [comment]
@mbasmanova I fixed those two errors. Can you try again? Thanks.

**Comment by mbasmanova** [comment]
@majetideepak Deepak, I'm working on updating Meta-specific connectors. Also, addressed a comment from @laithsakka to use const auto& [field, filter] when looping over subfieldFilters_:

```
    // Sort filters by subfield for deterministic output.
    std::map<std::string, common::Filter*> orderedFilters;
    for (const auto& [field, filter] : subfieldFilters_) {
      orderedFilters[field.toString()] = filter.get();
    }
    out << ", Filters: [";
    bool notFirstFilter = false;
    for (const auto& [field, filter] : orderedFilters) {
      if (notFirstFilter) {
        out << ", ";
      }
      out << "(" << field << ", " << filter->toString() << ")";
      notFirstFilter = true;
    }
    out << "]";
```

**Comment by majetideepak** [comment]
@mbasmanova thank you for handling this!

**Comment by mbasmanova** [comment]
@majetideepak This change broke presto_cpp, but @spershin quickly fixed it in https://github.com/facebookincubator/velox/pull/1288 and https://github.com/facebookexternal/presto_cpp/pull/679

Thank you, Sergey!

**Comment by majetideepak** [comment]
@spershin thanks for the fix.

**Pull request closed without merging**